[T7][W10-C4]

ZhangYiJiang opened the pull request.

damithc commented:
I have no further comments on the user guide. Try to get the functionality up sooner so that you can use it yourself and figure out how to fine tune UI and command format further.

damithc commented on collated/A0135805Hreused.md:
```diff
@@ -0,0 +1,351 @@
+    }
+
+    /**
+     * Configure the UI layout of {@link CommandFeedbackView}.
```
Configures

damithc commented on collated/A0135817Breuse.md:
```diff
@@ -0,0 +1,46 @@
+            return ((java.sql.Date) date).toLocalDate();
+        else
+            return Instant.ofEpochMilli(date.getTime()).atZone(zone).toLocalDate();
+    }
```
Coding standard violations in this method.

damithc commented on collated/A0139021U.md:
```diff
@@ -0,0 +1,790 @@
+    }
+
+    @Test
+    public void testPreviewAddCommand() throws InterruptedException {
```
You can skip `test` in the name because `@Test` annotation is already there.

damithc reviewed (commented):
Added a few comments for the collated code.

xiendong committed:
Abstract out method sequence for data validation

xiendong committed:
Perform method abstraction to TodoModel (tag segment)

xiendong committed:
Update code quality

xiendong committed:
Update code style to UniqueTagCollection

xiendong committed:
Code style update

xiendong committed:
Update imports

xiendong committed:
Fix GUI Test Cases

xiendong committed:
Update code style with SLAP

xiendong committed:
Further SLAP implementation for TagCommand

xiendong committed:
Extract rename validation check

cricketer94 committed:
Merge branch 'tagging_system_2' of https://github.com/CS2103AUG2016-W10-C4/main.git into tag-ui

xiendong committed:
Update code as per recommendation

xiendong committed:
Style update, makes completed dimmer, makes show more button darker

xiendong committed:
Fix styling with code

li-kai committed:
Merge pull request #158 from CS2103AUG2016-W10-C4/tagging_system_2

Tagging System 2

xiendong committed:
Update to place tags at the first row instead

xiendong committed:
Adjusted layout for title and pin

xiendong committed:
Update the way to get tag names in gui test

ZhangYiJiang committed:
attempt to reproduce issue #161

ZhangYiJiang committed:
Merge pull request #163 from CS2103AUG2016-W10-C4/tagging_system_2

UI Updates to Tags in Task Card View

ZhangYiJiang committed:
tweak command summary description text

li-kai committed:
Fix visual bugs

ZhangYiJiang committed:
link up IO exception event and todolist

ZhangYiJiang committed:
extract reused code to separate file

ZhangYiJiang committed:
Merge branch 'master' of github.com:CS2103AUG2016-W10-C4/main

cricketer94 committed:
Create Global Tag view pane

xiendong committed:
Remove tags from collapsing

cricketer94 committed:
Attach global tag view to main Window

cricketer94 committed:
Styling for globalTags

cricketer94 committed:
Add style to fix spaces between tags

cricketer94 committed:
Manage events related to global Tasks

cricketer94 committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main.git into tag-ui

Conflicts:
	src/main/resources/view/CommandPreviewView.fxml
	src/main/resources/view/MainWindow.fxml

ZhangYiJiang committed:
I came in like a wrecking ball
I never hit so hard in love
All I wanted was to break your walls
All you ever did was wreck me
Yeah, you, you wreck me

xiendong committed:
bug fixes

cricketer94 committed:
fix back command preview

ZhangYiJiang committed:
Merge pull request #164 from CS2103AUG2016-W10-C4/edit-command-161-test

Attempt to reproduce issue #161

xiendong committed:
Bug fixes with tagging system : delete and rename should be global

xiendong committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main into tag_system_4

# Conflicts:
#	src/main/java/seedu/todo/model/TodoList.java

ZhangYiJiang committed:
Merge pull request #166 from CS2103AUG2016-W10-C4/i-came-in-like-a-wreaking-ball

Cleaning up dead code

cricketer94 committed:
Fix merge conflicts

ZhangYiJiang committed:
Merge pull request #168 from CS2103AUG2016-W10-C4/tag_system_4

Tagging System: Disable Tag Collapsing and Bug Fixes

cricketer94 committed:
Fix Main Window merge conflict

cricketer94 committed:
Quick fix for only completed tasks to be affected by deleteAll

cricketer94 committed:
Fix styling errors :X

cricketer94 committed:
Remove unused import

cricketer94 committed:
Fix deleteAll() test bug

cricketer94 committed:
Add @@author tags

cricketer94 committed:
Fixed wrapping, padding and rename dismissal command

li-kai committed:
Add activity diagrams

xiendong committed:
Add code to allow search by tags

cricketer94 committed:
Change press back to type

cricketer94 committed:
add an additonal sort, such that completed events/tasks are lower down

li-kai committed:
Add tag partial match

li-kai committed:
Undo partial match

li-kai committed:
Merge pull request #169 from CS2103AUG2016-W10-C4/find_tags

Search by Tags

cricketer94 committed:
Change type back to press

cricketer94 committed:
Merge branch 'master' into multiple-indexed-commands-2. Conflicts merged

cricketer94 committed:
Merge branch 'master' into tag-ui

# Conflicts:
#	src/main/java/seedu/todo/logic/commands/TagCommand.java

cricketer94 committed:
Fixed unused imports

cricketer94 committed:
Merge pull request #165 from CS2103AUG2016-W10-C4/tag-ui

Adding view to show list of global tags

cricketer94 committed:
Merge pull request #159 from CS2103AUG2016-W10-C4/multiple-indexed-commands-2

Second implementation of indexed commands acting on many tasks

ZhangYiJiang committed:
ready app for v0.5 release

ZhangYiJiang committed:
Update README for v0.5

ZhangYiJiang committed:
Update README.md

ZhangYiJiang committed:
remove unnecessary capturing group

ZhangYiJiang committed:
increase sleep time in TodoTest to improve test consistency

ZhangYiJiang committed:
improve StringUtil edge case coverage

ZhangYiJiang committed:
Merge branch 'master' of github.com:CS2103AUG2016-W10-C4/main

ZhangYiJiang committed:
fix JavaFx thread exception in automatic TodoList event completion

ZhangYiJiang committed:
remove auto mark as complete for events

ZhangYiJiang committed:
fade overdue events out

ZhangYiJiang committed:
clean up imports

ZhangYiJiang committed:
rename splitString to split

ZhangYiJiang committed:
Merge pull request #175 from CS2103AUG2016-W10-C4/timer-bug-fix

Remove automarking event as complete

ZhangYiJiang committed:
clean up Model JavaDocs

cricketer94 committed:
Add clear command to ensure that clear command works

cricketer94 committed:
Improve find command with *some* feedback to tell user how to dismiss

cricketer94 committed:
Remove unused imports

cricketer94 committed:
Finish up implementation and testing for visual feedback via events

cricketer94 committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main.git into views-and-utils2

cricketer94 committed:
Improvements to view Utils

cricketer94 committed:
Modify sort to account for events which are over but not complete

ZhangYiJiang committed:
Merge pull request #177 from CS2103AUG2016-W10-C4/visible-user-feedback2

Visible user feedback

ZhangYiJiang committed:
make tags case insensitive

ZhangYiJiang committed:
update find command's tag flag

ZhangYiJiang committed:
Merge remote-tracking branch 'origin/master' into find-tags-improvement

# Conflicts:
#	src/main/java/seedu/todo/logic/commands/FindCommand.java

ZhangYiJiang committed:
Merge pull request #179 from CS2103AUG2016-W10-C4/find-tags-improvement

Find tags improvement

cricketer94 committed:
Fix test to follow new sorting order

cricketer94 committed:
Add tests for today. Tests not stable

cricketer94 committed:
Modifications to TimeUtil class and tests such that current time is
modifiable.

cricketer94 committed:
Fix broken tests

xiendong committed:
Added test to fix the undo with rename bug

cricketer94 committed:
Allow view to exit find results.

cricketer94 committed:
Edit feedback message

xiendong committed:
Reorganise the code

xiendong committed:
Shifting the validation method call to the helper methods

xiendong committed:
Add one more test to undo command - tag command test

xiendong committed:
Fixed bug #176

xiendong committed:
Add implementation for renaming a specific tag

xiendong committed:
Add command to command summary

xiendong committed:
Add tests for the new rename command

xiendong committed:
Update code quality

xiendong committed:
Check if TaskCardView is available. Log if not available

xiendong committed:
Merge branch 'scroll_and_select_bug' of https://github.com/CS2103AUG2016-W10-C4/main into tag_system_5

xiendong committed:
Created view handler for CommandErrorView

xiendong committed:
Add incorrect input test to add method

xiendong committed:
Add invalid input to delete test

xiendong committed:
Added test for help view

xiendong committed:
Update styling for GlobalTagView

xiendong committed:
Remove unused imports

xiendong committed:
Add view handle to GlobalTagView

xiendong committed:
Add global tag view tests

cricketer94 committed:
Merge branch 'scroll_and_select_bug' of https://github.com/CS2103AUG2016-W10-C4/main.git into find-command-bug

xiendong committed:
Add test for filter bar view

xiendong committed:
Merge branch 'scroll_and_select_bug' of https://github.com/CS2103AUG2016-W10-C4/main into gui_test_2

xiendong committed:
Add test to identify bug for isOverdueDisplayCorrect(...) check

xiendong committed:
Bug fixes

xiendong committed:
Merge branch 'scroll_and_select_bug' of https://github.com/CS2103AUG2016-W10-C4/main into overdue_test_bugfix

xiendong committed:
Add test to check for ongoing events

cricketer94 committed:
Merge branch 'overdue_test_bugfix' of https://github.com/CS2103AUG2016-W10-C4/main.git into find-command-bug

xiendong committed:
Merge branch 'overdue_test_bugfix' of https://github.com/CS2103AUG2016-W10-C4/main into gui_test_2

xiendong committed:
Add time delay fix to select

xiendong committed:
Merge branch 'overdue_test_bugfix' of https://github.com/CS2103AUG2016-W10-C4/main into scroll_and_select_bug

xiendong committed:
Shorten the bounded wait time

xiendong committed:
Merge branch 'overdue_test_bugfix' of https://github.com/CS2103AUG2016-W10-C4/main into tag_system_5

cricketer94 committed:
Merge branch 'scroll_and_select_bug' of https://github.com/CS2103AUG2016-W10-C4/main.git into find-command-bug

xiendong committed:
Added checks to global tag list in test cases

xiendong committed:
Resolve bug fix that global tag list does not get updated.

xiendong committed:
Remove sysout

li-kai committed:
Merge pull request #182 from CS2103AUG2016-W10-C4/scroll_and_select_bug

Scroll and Select Bug Fix

xiendong committed:
Add handle to search view status

xiendong committed:
Added test code for basic search test

xiendong committed:
Bug fixes to #162 on highlighting

xiendong committed:
Remove unused import

xiendong committed:
Resolves #183

xiendong committed:
Style fixes

li-kai committed:
Merge pull request #186 from CS2103AUG2016-W10-C4/event_tag_colouring

Task vs Event colouring: Resolves #183

xiendong committed:
Change type to press

li-kai committed:
Merge pull request #187 from CS2103AUG2016-W10-C4/event_tag_colouring

Change "type" to "press" in message to dismiss tag view

li-kai committed:
Merge pull request #181 from CS2103AUG2016-W10-C4/tag_system_5

Tags: Fix undo bug, and rename individual tags.

xiendong committed:
Style fixes

xiendong committed:
Remove given util function as requested

xiendong committed:
Remove unused import

Commit:
Merge pull request #185 from CS2103AUG2016-W10-C4/gui_test_2

Resolves #174

li-kai committed:
Add smart command to inputs

cricketer94 committed:
Merge branch 'master' into find-command-bug

xiendong committed:
Add fix to layout issues in some view panels

cricketer94 committed:
Add boolean parameter to event to disable toggling if required

cricketer94 committed:
Modify event handlers to handle boolean parameter

cricketer94 committed:
Change event posted by edit command

li-kai committed:
Add more tests for commands

cricketer94 committed:
Change feedback message

cricketer94 committed:
Add comment to describe what find(null) is for.

xiendong committed:
Update layout issue for help panel

xiendong committed:
Update layout issue for command error view

xiendong committed:
Typing a new character should clear the command message

xiendong committed:
Remove unused methods

li-kai committed:
Remove unused imports

xiendong committed:
Set min width.

li-kai committed:
Merge remote-tracking branch 'remotes/origin/master' into smart-parser

li-kai committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main into smart-parser

# Conflicts:
#	src/main/java/seedu/todo/logic/commands/CommandMap.java

li-kai committed:
Add clear command into tests

li-kai committed:
Optimize imports

cricketer94 committed:
Add requested changes

li-kai committed:
Satisfy codacy rules

cricketer94 committed:
Change test to make it stable

cricketer94 committed:
Edit feedback message again

cricketer94 committed:
Add more stabilised tests

li-kai committed:
Merge pull request #189 from CS2103AUG2016-W10-C4/smart-parser

Smart parser

li-kai committed:
Merge pull request #188 from CS2103AUG2016-W10-C4/view_panel_bugfix

Layout Bugfix: Solves the overlaying panels' layout issues.

xiendong committed:
Add initial implementation for empty list view

cricketer94 committed:
Fixed with suggestions implemented

xiendong committed:
Separating emptyListView from todoListView

xiendong committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main into empty_list_view

xiendong committed:
Reset changes to TaskViewFilter

xiendong committed:
Complete initial implementation for empty list view

cricketer94 committed:
Fix indentation and capitalization errors

li-kai committed:
Merge pull request #178 from CS2103AUG2016-W10-C4/views-and-utils2

Views and Utils

xiendong committed:
Update emoji for empty list

xiendong committed:
Update from master

xiendong committed:
complete implementation for empty list view

xiendong committed:
Update to icons

xiendong committed:
Merge branch 'scroll_and_select_bug_2' of https://github.com/CS2103AUG2016-W10-C4/main into empty_list_view

cricketer94 committed:
Merge branch 'scroll_and_select_bug_2' of https://github.com/CS2103AUG2016-W10-C4/main.git into highlight-task-bug2

xiendong committed:
update visuals

xiendong committed:
Add gui test for view command

xiendong committed:
Refine tag command

ZhangYiJiang committed:
remove completed todos and cleanup

xiendong committed:
Enforced new height condition

cricketer94 committed:
Ensure selections are cleared for tasks that have no visual feedback

ZhangYiJiang committed:
clean up Codacy issues

xiendong committed:
Gui updates

xiendong committed:
update image url

ZhangYiJiang committed:
fix today view test

ZhangYiJiang committed:
Update AboutUs.md

cricketer94 committed:
Remove string feedbacks for commands that show visible feedback.

ZhangYiJiang committed:
add github usernames

ZhangYiJiang committed:
clean up and tagging

xiendong committed:
Style updates

ZhangYiJiang committed:
clean up unused code

xiendong committed:
Further style fixes

ZhangYiJiang committed:
remove unused code

xiendong committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main into empty_list_view

cricketer94 committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main.git into highlight-task-bug2

cricketer94 committed:
Fix test bugs

xiendong committed:
add annotation

cricketer94 committed:
Fix broken test and address codacy review

cricketer94 committed:
Remove trailing space

cricketer94 committed:
Update About Us (Govind)

li-kai committed:
Remove StorageManager

xiendong committed:
Added details to about me

xiendong committed:
Update about me

li-kai committed:
Move initiation into UserPrefs

li-kai committed:
Update about me

li-kai committed:
Fix log message

li-kai committed:
Merge pull request #192 from CS2103AUG2016-W10-C4/simple-pref

Remove StorageManager

Commit:
Merge pull request #191 from CS2103AUG2016-W10-C4/empty_list_view

New feature: Empty List View, and many Ui Refinements

li-kai committed:
Merge pull request #180 from CS2103AUG2016-W10-C4/find-command-bug

Find command bug

xiendong committed:
Quick style update

xiendong committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main

ZhangYiJiang committed:
attempt to fix pin command race condition

ZhangYiJiang committed:
Merge branch 'highlight-task-bug2'

li-kai committed:
Fix storage instantiation

ZhangYiJiang committed:
Merge branch 'master' of github.com:CS2103AUG2016-W10-C4/main

ZhangYiJiang committed:
remove unused import

ZhangYiJiang committed:
fix missing @FXML annotation

ZhangYiJiang committed:
remove unused variable

li-kai committed:
Update diagrams

ZhangYiJiang committed:
update styles

ZhangYiJiang committed:
Merge remote-tracking branch 'origin/master'

ZhangYiJiang committed:
tag documentation files

ZhangYiJiang committed:
update user guide

li-kai committed:
Update diagrams

cricketer94 committed:
Add images for user guide

cricketer94 committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main.git

ZhangYiJiang committed:
add third-party libs

cricketer94 committed:
More images

li-kai committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main

li-kai committed:
Make png transparent

cricketer94 committed:
Update README with new image

cricketer94 committed:
Merge branch 'master' of https://github.com/CS2103AUG2016-W10-C4/main

ZhangYiJiang committed:
update user guide with grammar fixes

ZhangYiJiang committed:
Merge branch 'master' of github.com:CS2103AUG2016-W10-C4/main

ZhangYiJiang committed:
update dev guide

li-kai committed:
Fix collate comments

ZhangYiJiang committed:
update user guide page numebers

ZhangYiJiang committed:
Merge branch 'master' of github.com:CS2103AUG2016-W10-C4/main

li-kai committed:
Build collated

ZhangYiJiang committed:
Update README.md

ZhangYiJiang committed:
Update AboutUs.md
